Team 19 Project Proposal

nchintada opened the pull request.

nchintada committed:
Create proposal.md

bnklaiman committed:
Made names and others bullet points for legibility

nchintada committed:
Created base front end files

Created skeleton front end files and file structure to start being filled in.

nchintada committed:
Adding in base files for back-end

Added in package.json that calls the server.js file and a blank server.js file to be filled in.

lbodwell committed:
Update project structure. Add basic server. Add dependencies. Set environment variables.

lbodwell committed:
Set up API routing. Implement various Express middlewares. Set up db and add schemas for all collections. Implement basic socket.io interaction. Start GitHub OAuth.

charlieroberts commented:
This sounds like a good project, and is reasonable in scope. If you finish up early and want to consider a stretch goal, perhaps try adding voice (or video) chat via WebRTC with a library like https://github.com/feross/simple-peer#usage

Good luck!

rkayastha98 committed:
initial

rkayastha98 committed:
layout complete

rkayastha98 committed:
layout complete

rkayastha98 committed:
Merge pull request #1 from nchintada/rimsha-dev

Rimsha dev

bnklaiman committed:
Update home.html

nchintada committed:
Merge pull request #1 from nchintada/rimsha-dev

Rimsha dev

nchintada committed:
Merge pull request #2 from nchintada/home-page

Initial pr for home page

bnklaiman committed:
Changed second heading to groups

rkayastha98 committed:
layout complete

rkayastha98 committed:
Merge branch 'home-page' into rimsha-dev

rkayastha98 committed:
layout complete

lbodwell committed:
Minor tweaks.

nchintada committed:
Started writing task script

Currently the basic functions ported over from Rimsha's code work and I am setting up functionality for adding tasks on screen and to the database

lbodwell committed:
Work on user and group API routes.

lbodwell committed:
Fix error.

lbodwell committed:
Continue work on group routes.

lbodwell committed:
Finish user and group routes.

lbodwell committed:
User and group route improvements. Start message routes.

lbodwell committed:
Implement message routes. General cleanup.

lbodwell committed:
Fix bug in message routes. Add endpoints and comments for task routes.

nchintada committed:
Tasks can be added

Tasks can be added to the database and displayed on the column, now I need to change the display so that once a new task is added it will push down the add button. Next will be displaying all existing tasks from the database.

nchintada committed:
Adding and deleting column basic functionality works

We can now add and delete columns, extra code needs to be added to improve how it displays. We should probably also have a way to remove specific columns.

nchintada committed:
Merge branch 'master' into rimsha-dev

nchintada committed:
Merge pull request #3 from nchintada/rimsha-dev

Rimsha dev

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into master

nchintada committed:
Merge branch 'master' into nikhil-dev

nchintada committed:
Displays tasks on load

Now tasks are displayed upon the page loading (currently all tasks since I don't have code in for identifying which group it is). Also adjusted code a bit after the pull from master to make sure all previous functionality works as intended.

nchintada committed:
Updated new column and task display

Tasks now display before the button instead of around it. Added a div to enable this functionality. Also updated the addCol function so that when columns are generated they have a header and new task button.

nchintada committed:
New add task buttons now functional

Added code to make the new add task button generated by the new columns actually functional.

nchintada committed:
Merge pull request #4 from nchintada/nikhil-dev

Nikhil dev

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into master

bnklaiman committed:
worked on login page

rkayastha98 committed:
layout complete

nchintada committed:
Just a little commit so I can checkout another branch :)

Added a comment that I will keep in before checking out another branch to test stuff.

nchintada committed:
Merge branch 'rimsha-dev' into nikhil-dev

nchintada committed:
Rimsha's changes implemented

After merging Rimsha's changes I now have them working in tasks.js

nchintada committed:
Edit Modal Autopopulate

When you click on a task it now will auto populate the task's fields to the edit modal card

nchintada committed:
Editing Tasks Works

nchintada committed:
Deleting Tasks Works

Makes sure that the edit display doesn't open (this would happen since the delete buttons are part of the div containing each card). Deletes correct task from server and then updates page.

lbodwell committed:
Implement task routes.

lbodwell committed:
Resolve merge conflicts.

lbodwell committed:
Fix bug.

lbodwell committed:
Various bug fixes.

lbodwell committed:
Routing tweaks.

lbodwell committed:
Fix issue with account page.

nchintada committed:
Tasks now display on multiple columns

Added an additonal field called column to the tasks so they can keep track of what column they're on. Then added a number of modifications to existing functions to enable them to work with multiple columns.

lbodwell committed:
Fix bug preventing new users from being added to database.

nchintada committed:
Deleting Columns Works Now

Deleting columns functionality added in as well. If there are tasks in a specific column the user will be prompted if they wish to remove the tasks, if they decide all the tasks on that column will be deleted from the server.

nchintada committed:
Merging in task code from nikhil-dev

Should have all the working functionality: adding tasks, editing tasks, deleting tasks, adding columns, deleting columns and multi-column functionality

nchintada committed:
Merge branch 'master' into luke-dev

nchintada committed:
Merge pull request #6 from nchintada/luke-dev

API and GitHub Authentication

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into master

lbodwell committed:
Add groupId to Task schema.

lbodwell committed:
Merge branch 'luke-dev' of https://github.com/nchintada/final-project into luke-dev

lbodwell committed:
Merge pull request #7 from nchintada/luke-dev

Add groupId to Task schema

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into master

nchintada committed:
Merge branch 'master' into nikhil-dev

nchintada committed:
Starting to set up Task API functionality

Setting things up, committing now so I can checkout another branch.

nchintada committed:
Forgot this

rkayastha98 committed:
Merge remote-tracking branch 'origin/rimsha-dev' into rimsha-dev

# Conflicts:
#	public/js/tasks.js
#	public/tasks.html

nchintada committed:
Added server code

Forgot this, but the server code has been added.

rkayastha98 committed:
Merge branch 'nikhil-dev' into rimsha-dev

nchintada committed:
Updated again because it apparently missed the packages

nchintada committed:
Everything should work now

I think I finally actually got everything.

nchintada committed:
Commiting to test something

rkayastha98 committed:
Merge remote-tracking branch 'origin/rimsha-dev' into rimsha-dev

# Conflicts:
#	server.js

nchintada committed:
Slowly getting stuff working

Basically right now I have an idea of how to use Luke's API and column addition and deletion stuff works

rkayastha98 committed:
Merge branch 'nikhil-dev' into rimsha-dev

rkayastha98 committed:
layout complete

nchintada committed:
Have now successfully connected with API

Right now getting the groupId and getting the list of tasks works.

rkayastha98 committed:
Merge pull request #8 from nchintada/rimsha-dev

Rimsha dev

nchintada committed:
Setup for API based task functions

Waiting on update to date functionality from Luke. Once that is done, I can add in the proper steps to use the API.

nchintada committed:
Added the back button

nchintada committed:
Adding tasks works with API

Need to fix displaying tasks, addCol doesn't seem to be called properly

nchintada committed:
Displaying tasks works with API

Going to get tasks to display when added next.

nchintada committed:
Adding and displaying added tasks works with API

Moving to deleting tasks next

nchintada committed:
Deleting tasks works with API

Moving to editing next

bnklaiman committed:
sidebar appears again, but hamburger button still doesn't expand sidebar

bnklaiman committed:
updated login page

bnklaiman committed:
the cards now align

rkayastha98 committed:
layout complete

nchintada committed:
Editing Tasks works with API

Finally. I need to make some minor tweaks, but the core functionality is done.

rkayastha98 committed:
layout complete

acdesveaux committed:
lots of stuff

adds all tasks and groups, logout button works, fills user's display name in hellow message

acdesveaux committed:
Merge pull request #9 from nchintada/bnklaiman-dev

sidebar appears again, but hamburger button still doesn't expand sidebar

rkayastha98 committed:
Merge branch 'master' of https://github.com/nchintada/final-project into rimsha-dev

acdesveaux committed:
Merge pull request #10 from nchintada/bnklaiman-login-styling

updated login page

acdesveaux committed:
Merge branch 'master' into adam-dev

acdesveaux committed:
Merge branch 'nikhil-dev' into adam-dev

acdesveaux committed:
fixed server

fixed server

bnklaiman committed:
cleaned up login page

lbodwell committed:
Implement Google OAuth. Add invitation support to API. Various bug fixes.

lbodwell committed:
Resolve merge conflicts.

nchintada committed:
Merge pull request #12 from nchintada/adam-dev

Home js + Task js functionality

lbodwell committed:
Add back parts of server.js lost in merge.

lbodwell committed:
Actually add back everything lost in merge. Add functionality to login/logout buttons.

nchintada committed:
Merge branch 'master' into luke-dev

nchintada committed:
Merge pull request #13 from nchintada/luke-dev

Implement Google OAuth. Add invitation support to API. Various bug fixes.

lbodwell committed:
Minor tweaks.

lbodwell committed:
Resolve merge conflicts.

lbodwell committed:
Merge branch 'master' of https://github.com/nchintada/final-project into luke-dev

lbodwell committed:
Bug fixes.

lbodwell committed:
Merge pull request #14 from nchintada/luke-dev

Bug fixes.

bnklaiman committed:
added add group mini-form

nchintada committed:
Task functionality is working again

But there is a minor bug I think so I will work on fixing that

nchintada committed:
Merge branch 'master' of https://github.com/nchintada/final-project

acdesveaux committed:
add group button

add group button

bnklaiman committed:
added accept/reject buttons to invites

acdesveaux committed:
Merge branch 'master' into adam-dev

acdesveaux committed:
Merge pull request #16 from nchintada/adam-dev

add group button

bnklaiman committed:
test

bnklaiman committed:
Added accept and reject invite buttons

acdesveaux committed:
Merge pull request #17 from nchintada/bnklaiman-invite-accept

Added accept and reject invite buttons

nchintada committed:
Merge pull request #11 from nchintada/rimsha-dev

Rimsha dev

rkayastha98 committed:
layout complete

nchintada committed:
Merge branch 'master' into rimsha-dev

nchintada committed:
Merge pull request #18 from nchintada/rimsha-dev

Rimsha dev

lbodwell committed:
Chat testing.

bnklaiman committed:
improved chat window and delete button

lbodwell committed:
Persistent chat functionality.

acdesveaux committed:
add group works

add group works

acdesveaux committed:
Merge branch 'master' into adam-dev

acdesveaux committed:
Merge pull request #19 from nchintada/bnklaiman-chat

improved chat window and delete button

lbodwell committed:
Resolve merge conflicts.

lbodwell committed:
Merge pull request #21 from nchintada/luke-dev

Chat functionality

acdesveaux committed:
Merge pull request #20 from nchintada/adam-dev

add groups

lbodwell committed:
Bug fix.

lbodwell committed:
Merge pull request #22 from nchintada/luke-dev

Bug fix.

nchintada committed:
Have dropdown working and displaying properly

nchintada committed:
Now also shows up in the edit drop down

bnklaiman committed:
fixed chat window

lbodwell committed:
Fix bug with message routes.

rkayastha98 committed:
Update README.md

updated readme

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into bnklaiman-chat-window

lbodwell committed:
Add invitations route.

acdesveaux committed:
Merge pull request #23 from rkayastha98/master

Readme updated

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into bnklaiman-chat-window

acdesveaux committed:
Merge pull request #24 from nchintada/luke-dev

Add invitations route.

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into bnklaiman-chat-window

lbodwell committed:
Chat testing.

lbodwell committed:
More fixes.

lbodwell committed:
CSS tweaks.

lbodwell committed:
Merge branch 'master' of https://github.com/nchintada/final-project into luke-dev

nchintada committed:
Transferring group name between pages works

nchintada committed:
Delete buttons get added for each column

lbodwell committed:
Fix order of chat.

lbodwell committed:
Merge pull request #25 from nchintada/luke-dev

Chat fixes.

bnklaiman committed:
Merge branch 'master' of https://github.com/nchintada/final-project into bnklaiman-chat-window

bnklaiman committed:
added padding

nchintada committed:
Deleting columns back to original functionality

lbodwell committed:
Merge remote-tracking branch 'origin' into bnklaiman-chat-window

lbodwell committed:
Resolve merge conflicts.

lbodwell committed:
Resolve merge conflicts.

lbodwell committed:
Merge pull request #26 from nchintada/bnklaiman-chat-window

bnklaiman chat window

nchintada committed:
Merge branch 'master' into nikhil-dev

nchintada committed:
Header changes based on group name

lbodwell committed:
Merge branch 'master' into nikhil-dev

lbodwell committed:
Merge pull request #27 from nchintada/nikhil-dev

Adding in dropdown functionality, sending groupID between pages

nchintada committed:
Display for edit is also working now

nchintada committed:
Merge branch 'nikhil-dev' of https://github.com/nchintada/final-project into nikhil-dev

bnklaiman committed:
styled and improved chat text

nchintada committed:
Update README.md

nchintada committed:
Merge pull request #28 from nchintada/nikhil-dev

Edit display working also

lbodwell committed:
Add route for all users.

lbodwell committed:
Merge pull request #29 from nchintada/luke-dev

Add route for all users.

lbodwell committed:
Fix comment.

acdesveaux committed:
invite stuff not done

acdesveaux committed:
Merge branch 'master' into adam-dev

nchintada committed:
Inviting to a group works

lbodwell committed:
Minor tweaks.

lbodwell committed:
Merge pull request #30 from nchintada/luke-dev

Minor tweaks.

acdesveaux committed:
receiving invites works

acdesveaux committed:
Merge branch 'master' into adam-dev

acdesveaux committed:
Merge pull request #31 from nchintada/adam-dev

receiving invites works

nchintada committed:
Idk whats going on

nchintada committed:
Merge branch 'master' of https://github.com/nchintada/final-project

lbodwell committed:
Set callbacks for prod.

nchintada committed:
Update README.md
